consensus/propagation/catchup_test.go

```diff
@@ -0,0 +1,94 @@
+package propagation
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/stretchr/testify/assert"
+	proptypes "github.com/tendermint/tendermint/consensus/propagation/types"
+	cmtrand "github.com/tendermint/tendermint/libs/rand"
+	"github.com/tendermint/tendermint/types"
+)
+
+func TestCatchup(t *testing.T) {
+	reactors, _ := testBlockPropReactors(3)
+	reactor1 := reactors[0]
+	reactor2 := reactors[1]
+	reactor3 := reactors[2]
+
+	// setting the proposal for height 8 round 1
+	compactBlock := createCompactBlock(8, 1)
+	reactor1.AddProposal(compactBlock)
+	reactor2.AddProposal(compactBlock)
+	reactor3.AddProposal(compactBlock)
+
+	// setting the proposal for height 9 round 0
+	compactBlock = createCompactBlock(9, 1)
+	reactor1.AddProposal(compactBlock)
+	reactor2.AddProposal(compactBlock)
+	reactor3.AddProposal(compactBlock)
+
+	// setting the proposal for height 10 round 0
+	compactBlock = createCompactBlock(10, 0)
+	reactor1.AddProposal(compactBlock)
+	reactor2.AddProposal(compactBlock)
+	reactor3.AddProposal(compactBlock)
+
+	// setting the proposal for height 10 round 1
+	compactBlock = createCompactBlock(10, 1)
+	reactor1.AddProposal(compactBlock)
+	reactor2.AddProposal(compactBlock)
+	reactor3.AddProposal(compactBlock)
+
+	// setting the first reactor current height and round
+	reactor1.currentHeight = 8
+	reactor1.currentRound = 0
+
+	// handle the compact block
+	reactor1.handleCompactBlock(compactBlock, reactor1.self)
```

Comment:
would the current height being set to 8 every occur? what about handling the compact block with self as the sender but they aren't the proposer?